torch/csrc/autograd/engine.h

```diff
@@ -96,13 +95,17 @@ struct GraphTask {
 
   // Set an appropriate exception on this graph_task which was encountered while
   // running the provided function.
-  void set_exception(std::exception_ptr eptr, const std::shared_ptr<Node>& fn);
+  void set_exception(std::exception& e, const std::shared_ptr<Node>& fn);
 
   // Whether or not to stop execution for this GraphTask when an error is
   // encountered. When set to true, this would cause Engine::execute() to throw
   // an exception as soon as the autograd engine receives an exception.
   bool exit_on_error_;
 
+  // Future representing the completion of the graph task. Notified when all
+  // tasks are done.
+  std::shared_ptr<FutureVariableList> future_;
```

Comment:
can we call this something like `future_result_`? it's hard to tell what's holding inside this future from the var name.